src/App.tsx

```diff
@@ -1,56 +1,112 @@
-import React from 'react';
+import React, { useState, useMemo, useCallback } from 'react';
 import './App.scss';
+import cn from 'classnames';
 import { peopleFromServer } from './data/people';
 
+function debounce(callback: () => void,
+  delay: number): (param: string) => void {
+  const timerID = 0;
+
+  return () => {
+    window.clearTimeout(timerID);
+    window.setTimeout(() => {
+      callback();
+    }, delay);
+  };
+}
+
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [value, setValue] = useState<string>('');
+  const [name, setName] = useState<string | undefined>(' ');
+  const [born, setBorn] = useState<number | undefined>();
+  const [died, setDied] = useState<number | undefined>();
+  const [valueApplied, setValueApplied] = useState('');
+  const [focus, setFocus] = useState(false);
+  const applyValue = useCallback(
+    debounce(setValueApplied as () => void, 1000),
+    [valueApplied],
+  );
+
+  const onFocus = () => {
+    setFocus(true);
+  };
+
+  const handlerInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setValue(e.target.value);
+    applyValue(e.target.value);
+  };
+
+  const handlerItemChange = (person: string) => {
+    const findPerson = peopleFromServer.find((p) => p.name === person);
+
+    setValue(person);
+    setName(findPerson?.name);
+    setBorn(findPerson?.born);
+    setDied(findPerson?.died);
+  };
+
+  const filteredPosts = useMemo(() => {
+    return peopleFromServer
+      .filter((person) => person.name.toLowerCase()
+        .includes(valueApplied.toLowerCase()))
+      .map((person) => person);
+  }, [valueApplied]);
 
   return (
     <main className="section">
-      <h1 className="title">
-        {`${name} (${born} = ${died})`}
-      </h1>
+      {name !== ' ' ? (
+        <h1 className="title">
+          {`${name} (${born} = ${died})`}
+        </h1>
+      ) : (
+        <h1 className="title">No selected persons</h1>
+      )}
```

Comment:
h1 and class is the same, so use condition only for content inside

!name ? `${name} (${born} = ${died})` : 'No selected persons'